src/libraries/System.IO.Compression/tests/ZipArchive/zip_LargeFiles.cs

```diff
@@ -0,0 +1,48 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace System.IO.Compression.Tests
+{
+    [Collection(nameof(DisableParallelization))]
+    public class zip_LargeFiles : ZipFileTestBase
+    {
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsSpeedOptimized), nameof(PlatformDetection.Is64BitProcess))] // don't run it on slower runtimes
+        [OuterLoop("It requires almost 12 GB of free disk space")]
+        public static void UnzipOver4GBZipFile()
+        {
+            byte[] buffer = GC.AllocateUninitializedArray<byte>(1_000_000_000); // 1 GB
```

Comment:
Is there a reasonable chance this will OOM?  If so, consider catching that and skipping the test, rather than failing it.

Reply:
> Is there a reasonable chance this will OOM? 

It should be executed only in Outerloop runs for the 64 bit "fast runtimes" and not in parallel with other tests so I don't think it's needed (at least for now).

https://github.com/dotnet/runtime/blob/4979dff15273a2c33238a5d393f4895e9c94e538/src/libraries/System.IO.Compression/tests/ZipArchive/zip_LargeFiles.cs#L13

https://github.com/dotnet/runtime/blob/4979dff15273a2c33238a5d393f4895e9c94e538/src/libraries/System.IO.Compression/tests/ZipArchive/zip_LargeFiles.cs#L10

https://github.com/dotnet/runtime/blob/98a89c9d7444f219c7b1aeb3796525d35f04d540/src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs#L105-L106